library/src/main/java/com/novoda/downloadmanager/lib/FileDownloadInfo.java

```diff
@@ -454,4 +454,48 @@ private Long getLong(String column) {
             return cursor.getLong(cursor.getColumnIndexOrThrow(column));
         }
     }
+
+    static final class ControlStatus {
+
+        private int control;
+        private int status;
+
+        public ControlStatus(int control, int status) {
+            this.control = control;
+            this.status = status;
+        }
+
+        public boolean isPaused() {
+            return control == DownloadsControl.CONTROL_PAUSED;
+        }
+
+        public boolean isCanceled() {
+            return status == DownloadStatus.CANCELED;
+        }
+
+        static final class Reader {
+
+            private final ContentResolver contentResolver;
+            private final DownloadsUriProvider downloadsUriProvider;
+
+            public Reader(ContentResolver contentResolver, DownloadsUriProvider downloadsUriProvider) {
+                this.contentResolver = contentResolver;
+                this.downloadsUriProvider = downloadsUriProvider;
+            }
+
+            public ControlStatus newControlStatus(long id) {
+                String[] projection = {DownloadContract.Downloads.COLUMN_CONTROL, DownloadContract.Downloads.COLUMN_STATUS};
+                Uri uri = ContentUris.withAppendedId(downloadsUriProvider.getAllDownloadsUri(), id);
+                Cursor downloadsCursor = contentResolver.query(uri, projection, null, null, null);
+                try {
+                    downloadsCursor.moveToFirst();
+                    int control = downloadsCursor.getInt(0);
```

Comment:
The reason is because we do the projection so we don't need to query again to the cursor where are the columns, we have just done it at the projection line. As well is saves execution time for querying the column int when we have already specify them in the projection.